This is strange it looks like glslwaveformrenderersignal.h is skipped. It is IMHO skipped under the same conditions like the whole cpp file. 

I guess 
#include <QGLFramebufferObject>
defines QT_NO_OPENGL or QT_OPENGL_ES_2

hoverver, even more strange is 
```
scons: *** [lin32_build/src/waveform/renderers/glslwaveformrenderersignal.o] Error 1
In file included from src/waveform/renderers/glvsynctestrenderer.cpp:5:
src/waveform/renderers/glvsynctestrenderer.h:14:40: error: expected class-name before ‘{’ token
         protected QOpenGLFunctions_2_1 {
``` 
because there is nothing in between. 

Ideas?

--- a/src/waveform/renderers/glslwaveformrenderersignal.cpp
+++ b/src/waveform/renderers/glslwaveformrenderersignal.cpp
@@ -1,12 +1,9 @@
-#include <QtCore/qglobal.h>
-
+#include "waveform/renderers/glslwaveformrenderersignal.h"
 #if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
 #include <QGLFramebufferObject>
 
-#include "waveform/renderers/glslwaveformrenderersignal.h"
 #include "waveform/renderers/waveformwidgetrenderer.h"
-
 #include "waveform/waveform.h"
 #include "waveform/waveformwidgetfactory.h"
 

--- a/src/waveform/renderers/glslwaveformrenderersignal.h
+++ b/src/waveform/renderers/glslwaveformrenderersignal.h
@@ -1,13 +1,11 @@
 #pragma once
 
-#include <QtCore/qglobal.h>
-
+#include <QOpenGLFunctions_2_1>
 #if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
 #include <QGLFramebufferObject>
 #include <QGLShaderProgram>
 #include <QtOpenGL>
-#include <QOpenGLFunctions_2_1>
 
 #include "track/track.h"
 #include "util/memory.h"

--- a/src/waveform/renderers/glvsynctestrenderer.cpp
+++ b/src/waveform/renderers/glvsynctestrenderer.cpp
@@ -1,8 +1,5 @@
-#include <QtCore/qglobal.h>
-
-#if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
-
 #include "waveform/renderers/glvsynctestrenderer.h"
+#if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
 #include "waveform/renderers/waveformwidgetrenderer.h"
 #include "waveform/waveform.h"

--- a/src/waveform/renderers/glvsynctestrenderer.h
+++ b/src/waveform/renderers/glvsynctestrenderer.h
@@ -1,10 +1,7 @@
 #pragma once
 
-#include <QtCore/qglobal.h>
-
-#if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
-
 #include <QOpenGLFunctions_2_1>
+#if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
 #include "waveformrenderersignalbase.h"
 

--- a/src/waveform/renderers/glwaveformrendererfilteredsignal.cpp
+++ b/src/waveform/renderers/glwaveformrendererfilteredsignal.cpp
@@ -1,10 +1,8 @@
-#include <QtCore/qglobal.h>
-
+#include "glwaveformrendererfilteredsignal.h"
 #if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
 #include <QDomNode>
 
-#include "glwaveformrendererfilteredsignal.h"
 #include "track/track.h"
 #include "waveform/waveform.h"
 #include "waveformwidgetrenderer.h"

--- a/src/waveform/renderers/glwaveformrendererfilteredsignal.h
+++ b/src/waveform/renderers/glwaveformrendererfilteredsignal.h
@@ -1,11 +1,9 @@
 #pragma once
 
-#include <QtCore/qglobal.h>
-
+#include <QOpenGLFunctions_2_1>
 #if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
 #include <QDomNode>
-#include <QOpenGLFunctions_2_1>
 
 #include "waveformrenderersignalbase.h"
 

--- a/src/waveform/renderers/glwaveformrendererrgb.cpp
+++ b/src/waveform/renderers/glwaveformrendererrgb.cpp
@@ -1,8 +1,6 @@
-#include <QtCore/qglobal.h>
-
+#include "waveform/renderers/glwaveformrendererrgb.h"
 #if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
-#include "glwaveformrendererrgb.h"
 #include "waveformwidgetrenderer.h"
 #include "waveform/waveform.h"
 #include "waveform/waveformwidgetfactory.h"

--- a/src/waveform/renderers/glwaveformrendererrgb.h
+++ b/src/waveform/renderers/glwaveformrendererrgb.h
@@ -1,10 +1,7 @@
 #pragma once
 
-#include <QtCore/qglobal.h>
-
-#if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
-
 #include <QOpenGLFunctions_2_1>
+#if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
 #include "waveformrenderersignalbase.h"
 

--- a/src/waveform/renderers/glwaveformrenderersimplesignal.cpp
+++ b/src/waveform/renderers/glwaveformrenderersimplesignal.cpp
@@ -1,8 +1,6 @@
-#include <QtCore/qglobal.h>
-
+#include "waveform/renderers/glwaveformrenderersimplesignal.h"
 #if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
-#include "glwaveformrenderersimplesignal.h"
 #include "util/math.h"
 #include "waveform/waveform.h"
 #include "waveform/waveformwidgetfactory.h"

--- a/src/waveform/renderers/glwaveformrenderersimplesignal.h
+++ b/src/waveform/renderers/glwaveformrenderersimplesignal.h
@@ -1,10 +1,7 @@
 #pragma once
 
-#include <QtCore/qglobal.h>
-
-#if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
-
 #include <QOpenGLFunctions_2_1>
+#if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
 
 #include "waveformrenderersignalbase.h"
 

